packages/@aws-cdk/core/lib/asset-staging.ts

```diff
@@ -268,9 +268,19 @@ export class AssetStaging extends Construct {
    */
   private stageByCopying(): StagedAsset {
     const assetHash = this.calculateHash(this.hashType);
-    const stagedPath = this.stagingDisabled
-      ? this.sourcePath
-      : path.resolve(this.assetOutdir, renderAssetFilename(assetHash, path.extname(this.sourcePath)));
+    let stagedPath: string;
```

Comment:
Is it possible to refactor the implementation without using `let` and without the code becoming too difficult to read? You want to use `const` if at all possible.

Reply:
I could create a method that checks for double extensions and returns an extension string? This will replace `path.extname(this.sourcePath)`